### PR TITLE
Dat 4596 - mobile top leaderboard tests for pages w/o Infobox

### PR DESCRIPTION
**Pull request by nikodamn**
Add two tests which corresponds to move of Mobile Top Leaderboard ad on:
- pages with Infoboxes below the Infobox
- pages without Infoboxes below page header. 

Reviewers:
@Wikia/west-wing @drets @PiotrGackowski

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 1]
```diff
@@ -1,9 +1,13 @@
 package com.wikia.webdriver.testcases.adstests;
 
+import com.wikia.webdriver.common.core.Assertion;
 import com.wikia.webdriver.common.dataprovider.mobile.MobileAdsDataProvider;
 import com.wikia.webdriver.common.templates.mobile.MobileTestTemplate;
+import com.wikia.webdriver.elements.mercury.old.PortableInfoboxObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: MINOR Remove this unused import 'com.wikia.webdriver.elements.mercury.old.PortableInfoboxObject'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 2]
```diff
@@ -71,6 +73,46 @@ public void adsLeaderboardAndInContentOnPage(String wikiName,
   @Test(
       groups = "AdsSlotsMercury",
       dataProviderClass = MobileAdsDataProvider.class,
+      dataProvider = "leaderboardSlotOnPageWithInfobox"
+  )
+  public void adsLeaderboardOnPageWithInfobox(String wikiName,
+                                               String article,
```
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: alighment

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 3]
```diff
@@ -71,6 +73,46 @@ public void adsLeaderboardAndInContentOnPage(String wikiName,
...
+                                               String adUnit) {
+
+    String testedPage = urlBuilder.getUrlForPath(wikiName, article);
+    MobileAdsBaseObject ads = new MobileAdsBaseObject(driver, testedPage);
+
+    ads.verifyGptIframe(adUnit, MOBILE_TOP_LEADERBOARD, SRC);
+
+    Point leaderboardLocation = driver.findElement(By.id(MOBILE_TOP_LEADERBOARD)).getLocation();
+    Point infoboxLocation = driver.findElement(By.className("portable-infobox")).getLocation();
```
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Please don't use _low-level_ `findElement` in the test methods.
(one idea is to create `getElementLocationByCssSelector()` inside some `xxxObject` class and use it 4 times)

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 4]
```diff
@@ -71,6 +73,46 @@ public void adsLeaderboardAndInContentOnPage(String wikiName,
...
+                                              String article,
+                                              String adUnit) {
+
+    String testedPage = urlBuilder.getUrlForPath(wikiName, article);
+    MobileAdsBaseObject ads = new MobileAdsBaseObject(driver, testedPage);
+
+    ads.verifyGptIframe(adUnit, MOBILE_TOP_LEADERBOARD, SRC);
+
+    Point leaderboardLocation = driver.findElement(By.id(MOBILE_TOP_LEADERBOARD)).getLocation();
+    Point pageHeader = driver.findElement(By.className("wiki-page-header")).getLocation();
```
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Consider moving `wiki-page-header` to static var

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 5]
```diff
@@ -71,6 +73,46 @@ public void adsLeaderboardAndInContentOnPage(String wikiName,
...
+
+    Assertion.assertTrue(leaderboardLocation.getY() > pageHeader.getY());
```
Comment (SebastianMarzjan) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Per offline conversation, it makes sense to compare TOP position of leaderboard with BOTTOM position of the header

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 6]
```diff
@@ -71,6 +73,46 @@ public void adsLeaderboardAndInContentOnPage(String wikiName,
...
+
+    String testedPage = urlBuilder.getUrlForPath(wikiName, article);
+    MobileAdsBaseObject ads = new MobileAdsBaseObject(driver, testedPage);
+
+    ads.verifyGptIframe(adUnit, MOBILE_TOP_LEADERBOARD, SRC);
+
+    Point leaderboardLocation = driver.findElement(By.id(MOBILE_TOP_LEADERBOARD)).getLocation();
+    Point infoboxLocation = driver.findElement(By.className("portable-infobox")).getLocation();
+
+    Assertion.assertTrue(leaderboardLocation.getY() > infoboxLocation.getY());
```
Comment (SebastianMarzjan) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Per offline conversation, it makes sense to compare TOP position of leaderboard with BOTTOM position of the infobox

**Comment by PiotrGackowski** [comment]
Please consider using 
  @InBrowser(
      browser = Browser.CHROME,
      emulator = Emulator.GOOGLE_NEXUS_5
  )

**Comment by drets** [comment]
Just thoughts: do we really need those tests?
(the feature is tested by [article-content-test.js](https://github.com/Wikia/mercury/blob/dev/front/main/tests/unit/components/article-content-test.js))

**Comment by dianafa** [comment]
@drets I think we still do, those you mentioned are unit ones, and it would be great to have coverage of this feature also in selenium tests.

**Comment by dianafa** [comment]
Also, are we checking the case, when the infobox is so long that is outside the viewport? I would test it in the scope of `adsLeaderboardOnPageWithInfobox`  - only to prepare infobox that is long enough to push ad below viewport. What do you think?

**Comment by drets** [comment]
Sounds good :wink:

**Comment by nikodamn** [comment]
@dianafa yup, currently Infobox has greater height then viewport, so tests is checking the location of ad not in the first viewport.

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java (hunk 2 above)
Comment (nikodamn) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Hey, what's wrong with this alingment? Is it about space before second parameter? I'm not familiar with this convention and I used it as it's on other tests in this class. Thanks!

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java [hunk 7]
```diff
@@ -59,6 +59,7 @@
 import org.openqa.selenium.Dimension;
 import org.openqa.selenium.JavascriptExecutor;
 import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.Point;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java]: MINOR Remove this unused import 'org.openqa.selenium.Point'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by nikodamn** [comment]
@drets @PiotrGackowski @Wikia/west-wing

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 8]
```diff
@@ -12,6 +16,9 @@
   private static final String MOBILE_IN_CONTENT = "MOBILE_IN_CONTENT";
   private static final String MOBILE_PREFOOTER = "MOBILE_PREFOOTER";
   private static final String SRC = "mobile";
+  private static final String PORTABLE_INFOBOX = ".portable-infobox";
+  private static final String ARTICLE_HEADER = ".wiki-page-header";
+  private static final String MOBILE_TOP_LEADERBOARD_ID = "#MOBILE_TOP_LEADERBOARD";
```
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Could you use [AdsContent.java](https://github.com/Wikia/selenium-tests/blob/master/src/test/java/com/wikia/webdriver/common/contentpatterns/AdsContent.java)?

E.g:  
`AdsContent.MOBILETOP_LB` for getting slot name   
or  
`AdsContent.getSlotSelector(AdsContent.MOBILETOP_LB)` for getting slot selector

File: src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java [hunk 9]
```diff
@@ -68,6 +75,59 @@ public void adsLeaderboardAndInContentOnPage(String wikiName,
...
+    PortableInfobox infobox = new PortableInfobox();
+
+    ads.verifyGptIframe(adUnit, MOBILE_TOP_LEADERBOARD, SRC);
+
+    int infoboxYPosition = infobox.getElementLocationByCssSelector(PORTABLE_INFOBOX).getY();
+    int infoboxHeight = infobox.getElementSizeByCssSelector(PORTABLE_INFOBOX).getHeight();
+    int leaderboardTopYPosition =
+        ads.getElementLocationByCssSelector(MOBILE_TOP_LEADERBOARD_ID).getY();
+
+    Assertion.assertTrue(leaderboardTopYPosition >= (infoboxHeight + infoboxYPosition));
```
Comment (drets) [inline src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsSlotsMercury.java]: Maybe `>`?

**Comment by drets** [comment]
Minor comments, otherwise LGTM :+1:

**Comment by idradm** [comment]
LGTM

**Pull request merged**